posthog/admin/admins/user_admin.py

```diff
@@ -124,4 +137,34 @@ def change_view(self, request, object_id, form_url="", extra_context=None):
             # Redirect back to the change form
             return HttpResponseRedirect(request.path)
 
+        if request.POST.get("revoke_sessions") == "1":
+            try:
+                user = self.get_object(request, object_id)
+                if user:
+                    num_revoked = self.delete_user_sessions(user)
+                    messages.success(request, f"Revoked {num_revoked} session(s)")
+                else:
+                    messages.warning(request, "User not found.")
+            except Exception as e:
+                messages.error(request, f"Failed to revoke sessions: {str(e)}")
+
+            # Redirect back to the change form
+            return HttpResponseRedirect(request.path)
+
         return super().change_view(request, object_id, form_url, extra_context)
+
+    def _user_sessions(self, user):
+        """Fetch user's active sessions. Uses an iterator due to large table size and lack of effective indexing."""
+        user_pk = str(user.pk)
+        sessions = Session.objects.filter(expire_date__gt=timezone.now()).only("session_data")
```

Comment:
In prod US this query currently returns ~50k rows

Reply:
Unfortunately no, the `django_session` table is unnecessarily obtuse.

<img width="810" height="115" alt="Screenshot 2025-07-31 at 09 40 52" src="https://github.com/user-attachments/assets/7d49006f-fbc5-4205-b6be-9d2ff1faf4f6" />

I'm assuming you're referring to the http response timeout? What's it currently set to in prod? (struggling to find the relevant code)

Reply:
Ah I see - didn't realize there could be such broad performance implications. Infra is toying with the idea of moving Admin to its own pods, so this would be another good use case for that. Worth noting that I expect us to use this functionality very infrequently.